scrapy/utils/conf.py

```diff
@@ -34,6 +34,16 @@ def _map_keys(compdict):
             _check_components(compdict)
             return {convert(k): v for k, v in six.iteritems(compdict)}
 
+    def _ensure_numeric_value(compdict):
+        for name, value in six.iteritems(compdict):
+            if value is None or isinstance(value, (int, float)):
+                continue
+            try:
+                compdict[name] = float(value)
```

Comment:
Allow to continue if there is a numeric value inside a string, like `" 4 "` or `\n8\n`. That would have played nice with the code referenced in #2420, but I don't really have a strong opinion about it.